src/Microsoft.Extensions.Http/DefaultHttpClientFactory.cs

```diff
@@ -117,15 +117,9 @@ internal class DefaultHttpClientFactory : IHttpClientFactory
 
         public HttpClient CreateClient(string name)
         {
-            if (name == null)
-            {
-                throw new ArgumentNullException(nameof(name));
-            }
-
-            var entry = _activeHandlers.GetOrAdd(name, _entryFactory).Value;
-            var client = new HttpClient(entry.Handler, disposeHandler: false);
+            HttpMessageHandler handler = CreateHandler(name);
```

Comment:
We should still null-check here, generally we validate arguments as close as possible to user code (even if it's redundant).

Also needs some var.